pkg/manager/manager.go

```diff
@@ -320,25 +336,22 @@ func (s *FraktiManager) RemoveContainer(ctx context.Context, req *kubeapi.Remove
 // ListContainers lists all containers by filters.
 func (s *FraktiManager) ListContainers(ctx context.Context, req *kubeapi.ListContainersRequest) (*kubeapi.ListContainersResponse, error) {
 	glog.V(3).Infof("ListContainers with request %s", req.String())
-	var (
-		containers []*kubeapi.Container
-		err        error
-	)
-	// kubelet always ant to list all containers, it use filter to get what it want
-	vmContainers, err := s.hyperRuntimeService.ListContainers(req.GetFilter())
-	if err != nil {
-		glog.Errorf("ListContainers from hyper runtime service failed: %v", err)
-		return nil, err
-	}
-	containers = append(containers, vmContainers...)
 
-	if s.privilegedRuntimeService != nil {
-		osContainers, err := s.privilegedRuntimeService.ListContainers(req.GetFilter())
+	var containers []*kubeapi.Container
+	for _, runtimeService := range []runtime.RuntimeService{
+		s.hyperRuntimeService,
+		s.privilegedRuntimeService,
+		s.unikernelRuntimeService,
```

Comment:
Could we add those runtime services to a list and refer to it when required? (same with other places)

Reply:
Only non-nil runtimes should be kept

Reply:
Yep, that's great idea.